metrics/web/test/common/presentation/navigation/state/navigation_notifier_test.dart

```diff
@@ -247,6 +272,202 @@ void main() {
       },
     );
 
+    test(
+      ".handlePageParametersUpdates() does not change the current page parameters if the given page parameters model is null",
+      () {
+        when(pageParametersFactory.create(any)).thenReturn(pageParametersModel);
+
+        notifier.handleAuthenticationUpdates(isLoggedIn: true);
+        notifier.push(DefaultRoutes.dashboard);
+
+        final initialPageParameters = notifier.currentPageParameters;
+
+        notifier.handlePageParametersUpdates(null);
+
+        expect(notifier.currentPageParameters, equals(initialPageParameters));
+      },
+    );
+
+    test(
+      ".handlePageParametersUpdates() does not change the current route configuration if the given page parameters model is null",
+      () {
+        notifier.handleAuthenticationUpdates(isLoggedIn: true);
+        notifier.push(DefaultRoutes.dashboard);
+
+        final currentConfiguration = notifier.currentConfiguration;
+
+        notifier.handlePageParametersUpdates(null);
+
+        expect(notifier.currentConfiguration, equals(currentConfiguration));
+      },
+    );
+
+    test(
+      ".handlePageParametersUpdates() does not update the last page if the given page parameters model is null",
+      () {
+        notifier.handleAuthenticationUpdates(isLoggedIn: true);
+        notifier.push(DefaultRoutes.dashboard);
+
+        final lastPage = notifier.pages.last;
+
+        notifier.handlePageParametersUpdates(null);
+
+        expect(notifier.pages.last, equals(lastPage));
```

Comment:
Should we check the length of the pages to make sure that the list is not changed also?